src/parser/core/modules/GlobalCooldown.js

```diff
@@ -95,66 +126,77 @@ export default class GlobalCooldown extends Module {
 		})
 	}
 
-	saveGcd(event, isInstant) {
-		let gcdLength = -1
+	//saveGcd(event, isInstant) {
+	saveGcd(gcdInfo, timestamp) {
+		if (!gcdInfo.event) {
+			return
+		}
+
+		const action = getAction(gcdInfo.event.ability.guid)
+		let speedMod = this.speedmod.get(gcdInfo.event.timestamp)
+		let castTime = action.castTime
 
-		if (this._lastGcd >= 0) {
-			// GCD is only to two decimal places, so round it there. Storing in Ms.
-			// eslint-disable-next-line no-magic-numbers
-			gcdLength = Math.round((event.timestamp - this._lastGcd)/10)*10
+		// HACK NOTE TODO: Need to properly account for abilities that alter only the cast or recast of attacks.
+		// Thinking of moving this into a module like speedmod, that can be called with a timestamp to grab modified base castTime/cooldown values
```

Comment:
Astral/umbral handling is technically supposed to be handled by `castTime`, though I recognise the current framework in there doesn't actually support what you need to do.

I'll let this pass for now despite the 🤢 , but ping me on deeskerd or w/e to chat about potential impl resolutions if you'd like.